vowpalwabbit/csoaa.cc

```diff
@@ -137,13 +189,15 @@ base_learner* csoaa_setup(VW::setup_base_i& stack_builder)
   VW::workspace& all = *stack_builder.get_all_pointer();
   auto c = VW::make_unique<csoaa>();
   option_group_definition new_options("Cost Sensitive One Against All");
-  new_options.add(
-      make_option("csoaa", c->num_classes).keep().necessary().help("One-against-all multiclass with <k> costs"));
+  new_options
+      .add(make_option("csoaa", c->num_classes).keep().necessary().help("One-against-all multiclass with <k> costs"))
+      .add(make_option("indexing", c->indexing).one_of({0, 1}).keep().help("Choose between 0 or 1-indexing"));
```

Comment:
should this be `necessary`? what happens if it isn't supplied?

Reply:
It will auto-detect if the flag isnt supplied. Eg if k=4 and 4 is seen it will become 1-indexed, if 0 is seen it will become 0-indexed, which effects how warnings are given for future mislabels

Reply:
should we output an `[info]` msg with the auto indexing chosen?

Reply:
Yea good idea -- added it for oaa too